TheMET/TheMET/LoadingPlaceholderView/LoadingPlaceholderView.swift

```diff
@@ -27,22 +28,21 @@ class LoadingPlaceholderView: UIView {
     
     override init(frame: CGRect) {
         super.init(frame: frame)
+        self.container.translatesAutoresizingMaskIntoConstraints = false
         self.animationView.translatesAutoresizingMaskIntoConstraints = false
         self.textLabel.translatesAutoresizingMaskIntoConstraints = false
-        self.addSubview(self.animationView)
-        self.addSubview(self.textLabel)
+        self.addSubview(self.container)
+        self.container.addArrangedSubview(self.animationView)
+        self.container.addArrangedSubview(self.textLabel)
+        self.container.axis = .vertical
+        self.container.distribution = .equalCentering
+        self.container.alignment = .center
         NSLayoutConstraint.activate([
-            self.animationView.heightAnchor.constraint(equalTo: self.heightAnchor, multiplier: 0.6),
-            self.animationView.widthAnchor.constraint(equalTo: self.widthAnchor, multiplier: 0.5),
-            self.animationView.centerXAnchor.constraint(equalTo: self.centerXAnchor),
-            self.animationView.topAnchor.constraint(equalTo: self.topAnchor, constant: 30),
+            self.container.heightAnchor.constraint(equalTo: self.heightAnchor, multiplier: 0.6),
```

Comment:
Specifying width&height for the container means that animation-view size will be different depending on the size of text in text-label.
I suggest to keep animation-view height & width to be bound to self width and height